File: src/CommonLetters.java
import java.util.*;

public class CommonLetters {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        List<String> words = new ArrayList<>();

        // Read words from the input
        while (scanner.hasNext()) {
            words.add(scanner.next());
        }

        // Find the common characters
        Set<Character> commonChars = getCommonCharacters(words);

        // Sort the characters in alphabetical order
        List<Character> sortedChars = new ArrayList<>(commonChars);
        Collections.sort(sortedChars);

        // Print the sorted common characters without spaces
        for (char c : sortedChars) {
            System.out.print(c);
        }
    }

    static Set<Character> getCommonCharacters(List<String> words) {
        // Initialize the set with characters from the first word
        Set<Character> commonChars = new HashSet<>();
        for (char c : words.get(0).toCharArray()) {
            commonChars.add(c);
        }

        // Find the intersection of all sets
        for (int i = 1; i < words.size(); i++) {
            Set<Character> wordChars = new HashSet<>();
            for (char c : words.get(i).toCharArray()) {
                wordChars.add(c);
            }
            commonChars.retainAll(wordChars);
        }

        return commonChars;
    }
}

File: src/common_letters.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	words := readWords()

	// Find the common characters
	commonChars := getCommonCharacters(words)

	// Sort and print the characters in alphabetical order without spaces
	for c := range commonChars {
		fmt.Print(string(c))
	}
}

func readWords() []string {
	var words []string
	scanner := bufio.NewScanner(os.Stdin)

	// Read words from the input
	for scanner.Scan() {
		words = append(words, scanner.Text())
	}

	return words
}

func getCommonCharacters(words []string) map[rune]bool {
	// Initialize the map with characters from the first word
	commonChars := make(map[rune]bool)
	for _, c := range words[0] {
		commonChars[c] = true
	}

	// Find the intersection of all maps
	for _, word := range words[1:] {
		wordChars := make(map[rune]bool)
		for _, c := range word {
			wordChars[c] = true
		}

		// Remove characters not present in the current word
		for c := range commonChars {
			if !wordChars[c] {
				delete(commonChars, c)
			}
		}
	}

	return commonChars
}
